docs/gen-ai/gen-ai-events.md

```diff
@@ -0,0 +1,383 @@
+| `content`    | `AnyValue`                     | The contents of the assistant message.                          | `Spans, events, metrics defined by the GenAI semantic conventions.` | `Opt-In` |
+| `tool_calls` | [ToolCall](#toolcall-object)[] | The tool calls generated by the model, such as function calls.  | `[{"id":"call_mszuSIzqtI65i1wAUOE8w5H4", "function":{"name":"get_link_to_otel_semconv", "arguments":{"semconv":"gen_ai"}}, "type":"function"}]` | `Conditionally Required`: if available |
+
+### `ToolCall` object
+
+| Body Field | Type                        | Description                        | Examples                                        | [Requirement Level](https://opentelemetry.io/docs/specs/semconv/general/attribute-requirement-level/) |
+|------------|-----------------------------|------------------------------------|-------------------------------------------------|-------------------|
+| `id`       | string                      | The id of the tool call            | `call_mszuSIzqtI65i1wAUOE8w5H4`                 | `Required`        |
+| `type`     | string                      | The type of the tool               | `function`                                      | `Required`        |
+| `function` | [Function](#function-object)| The function that the model called | `{"name":"get_link_to_otel_semconv", "arguments":{"semconv":"gen_ai"}}` | `Required` |
+
+### `Function` object
+
+| Body Field  | Type       | Description                            | Examples                   | [Requirement Level](https://opentelemetry.io/docs/specs/semconv/general/attribute-requirement-level/) |
+|-------------|------------|----------------------------------------|----------------------------|-------------------|
+| `name`      | string     | The name of the function to call       | `get_link_to_otel_semconv` | `Required`        |
+| `arguments` | `AnyValue` | The arguments to pass the the function | `{"semconv": "gen_ai"}`    | `Opt-In`          |
+
+## Tool event
+
+This event describes the output of the tool or function submitted back to the model.
+
+The event name MUST be `gen_ai.tool.message`.
+
+| Body Field     | Type   | Description                                   | Examples                        | [Requirement Level](https://opentelemetry.io/docs/specs/semconv/general/attribute-requirement-level/) |
+|----------------|--------|-----------------------------------------------|---------------------------------|-------------------|
+| `role`         | string | The actual role of the message author as passed in the message. | `"tool"`, `"function"` | `Conditionally Required`: if available and if not equal to `tool` |
+| `content`      | string | The contents of the tool message.             | `opentelemetry.io`              | `Opt-In`          |
```

Comment:
For my understanding, why is `content` (in Tool event) not `AnyValue` (for consistency with "User event" and "Assistant event")?

Reply:
good point, let's convert it to AnyValue too so it can be extended in the future